distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/KadAction.java

```diff
@@ -0,0 +1,348 @@
+        this.actionPeer = smsPeer;
+        this.actionType = actionType;
+        this.operationId = id;
+        this.currentPart = part;
+        this.totalParts = maxParts;
+        this.payloadType = payloadType;
+        this.payload = payload;
+        if (!isValid())
+            throw new IllegalArgumentException();
+    }
+
+    /**
+     * Constructor of KadDction given SMSMessage.
+     *
+     * @param buildingMessage the given SMSMessage.
+     * @throws IllegalArgumentException if the parameters are not valid.
+     */
+    public KadAction(@NonNull SMSMessage buildingMessage) {
+        String messageBody = buildingMessage.getData();
+        String[] parameteres = messageBody.split(SEPARATOR);
+        actionPeer = buildingMessage.getPeer();
+        if (parameteres.length == TOTAL_PARAMS) {
+            try {
+                actionType = ActionType.getTypeFromVal(Integer.parseInt(parameteres[ACTION_TYPE_POSITION]));
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException(e.getMessage() + "\n" + ACTION_CODE_NOT_FOUND_ERROR_MSG);
+            }
+
+            try {
+                operationId = Integer.parseInt(parameteres[OPERATION_ID_POSITION]);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException(e.getMessage() + "\n" + ID_NOT_FOUND_ERROR_MSG);
+            }
+
+            try {
+                totalParts = Integer.parseInt(parameteres[TOTAL_MESSAGES_POSITION]);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException(e.getMessage() + "\n" + TOTAL_MESSAGES_NOT_FOUND_ERROR_MSG);
+            }
+            try {
+                currentPart = Integer.parseInt(parameteres[CURRENT_MESSAGE_POSITION]);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException(e.getMessage() + "\n" + CURRENT_MESSAGE_NOT_FOUND_ERROR_MSG);
+            }
+            try {
+                payloadType = PayloadType.getTypeFromCode(Integer.parseInt(parameteres[PAYLOAD_TYPE_POSITION]));
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException(e.getMessage() + "\n" + PAYLOAD_TYPE_NOT_FOUND_ERROR_MSG);
+            }
+            payload = parameteres[PAYLOAD_POSITION];
+            if (!isValid())
+                throw new IllegalArgumentException();
+        }
+    }
+
+    /**
+     * Check if all the action's parameters are valid
+     *
+     * @return True if the defined action is valid and fits into a Message.
+     */
+    @Override
+    public boolean isValid() {
+        if (this.actionType.equals(ActionType.INVALID))
+            return false;
+        if (this.payloadType.equals(PayloadType.INVALID))
+            return false;
+        if (currentPart <= 0 || totalParts <= 0 || currentPart > totalParts)
```

Comment:
As suggested below I'd put a static field min = 0 and max = 99, valid for operationId, totalParts and currentParts, so we know it's 2 chars each, if you do so, isValid should check the values are within the bounds

Reply:
100 is too little in my opinion: if you generate id as random, the probability of colliding with another id generated by another peer is 1%. Let's use at least index of 0-999.